src/machine.rs

```diff
@@ -478,7 +478,27 @@ impl<'mir, 'tcx> Machine<'mir, 'tcx> for Evaluator<'mir, 'tcx> {
                 (None, Tag::Untagged)
             };
         let race_alloc = if let Some(data_race) = &memory_extra.data_race {
-            Some(data_race::AllocExtra::new_allocation(&data_race, alloc.size))
+            match kind {
```

Comment:
The `machine` file is meant to mostly be doing some gluing. This here is some proper "business logic". Could you move this inside the `data_race::AllocExtra::new_allocation` constructor?